### PR TITLE
Fix Silver-Mueller boundary condition in 1D

### PR DESCRIPTION
**Pull request by RemiLehe**
Thanks to @ax3l for spotting this and helping me fix this!

### CODE REVIEW

File: Source/FieldSolver/FiniteDifferenceSolver/ApplySilverMuellerBoundary.cpp [hunk 1]
```diff
@@ -190,9 +190,12 @@ void FiniteDifferenceSolver::ApplySilverMuellerBoundary (
     bool const apply_hi_y = (field_boundary_hi[1] == FieldBoundaryType::Absorbing_SilverMueller);
     bool const apply_lo_z = (field_boundary_lo[2] == FieldBoundaryType::Absorbing_SilverMueller);
     bool const apply_hi_z = (field_boundary_hi[2] == FieldBoundaryType::Absorbing_SilverMueller);
-#else
+#elif WARPX_DIM_XZ
     bool const apply_lo_z = (field_boundary_lo[1] == FieldBoundaryType::Absorbing_SilverMueller);
     bool const apply_hi_z = (field_boundary_hi[1] == FieldBoundaryType::Absorbing_SilverMueller);
+#else
+    bool const apply_lo_z = (field_boundary_lo[0] == FieldBoundaryType::Absorbing_SilverMueller);
```
Comment (dpgrote) [inline Source/FieldSolver/FiniteDifferenceSolver/ApplySilverMuellerBoundary.cpp]: Could this use the macro `WARPX_ZINDEX`?

**Comment by EZoni** [comment]
Could we use `WARPX_ZINDEX` to avoid this kind of bugs?

Like having these lines only once, for all dimensions:
```cpp
    bool const apply_lo_z = (field_boundary_lo[WARPX_ZINDEX] == FieldBoundaryType::Absorbing_SilverMueller);
    bool const apply_hi_z = (field_boundary_hi[WARPX_ZINDEX] == FieldBoundaryType::Absorbing_SilverMueller);
```

Update: never mind, matches what Dave suggested above.

**Comment by anecas1979** [comment]
Issue #3704 is resolved by this PR so we can close.  Quick note: a pulse with 1000x smaller amplitude is reflected.  Thank you so much.

**Comment by RemiLehe** [comment]
@dpgrote @EZoni That's a good point, thanks! I added this.

**Comment by RemiLehe** [comment]
@dpgrote @EZoni I just added an automated test ; feel free to merge the PR if you agree with the changes.

**Review by dpgrote — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**